src/main/java/com/faforever/client/chat/ChannelTabController.java

```diff
@@ -183,6 +207,24 @@ private void updateChannelTopic() {
             }
           });
     }
+    topicText.setVisible(notBlank);
+  }
+
+  private void updateChannelTopic() {
+    String oldTopicContent = topicContent;
+    ChannelTopic topic = chatChannel.getTopic();
+    setChannelTopic(topic.getContent());
+    if (StringUtils.isNotBlank(topic.getAuthor())) {
+      onChatMessage(new ChatMessage(channelName, Instant.now(), topic.getAuthor(), i18n.get("chat.topicUpdated")));
+      onChatMessage(new ChatMessage(channelName, Instant.now(), topic.getAuthor(), i18n.get("chat.oldTopicContent", oldTopicContent)));
+      onChatMessage(new ChatMessage(channelName, Instant.now(), topic.getAuthor(), i18n.get("chat.newTopicContent", topic.getContent())));
```

Comment:
This should all be sent as a single message so that it doesn't get broken up by other messages